From 98f8fa2cd787adb4f7887f2821b0ff44b823f74a Mon Sep 17 00:00:00 2001
From: Javier Guerra Melgares <javier.guerra@dekra.com>
Date: Tue, 25 Jun 2019 09:39:54 +0200
Subject: [PATCH] Remove credential by credid

Remove a credential on the own device by the "credid" attribute.

Signed-off-by: Javier Guerra Melgares <javier.guerra@dekra.com>
Reviewed-by: Larry Sachs <larry.j.sachs@intel.com>
Reviewed-by: Rick Bell <richard.s.bell@intel.com>
Reviewed-by: George Nash <george.nash@intel.com>
Reviewed-by: Kishen Maloor <kishen.maloor@intel.com>
Change-Id: I7fcab1ac67d6cd4be368bc0984ecc33a0bd5e0ce
---
 include/oc_obt.h  |  2 ++
 security/oc_obt.c | 11 +++++++++++
 2 files changed, 13 insertions(+)

diff --git a/include/oc_obt.h b/include/oc_obt.h
index 5fa0b87..95e6a2d 100644
--- a/include/oc_obt.h
+++ b/include/oc_obt.h
@@ -106,6 +106,8 @@ int oc_obt_provision_ace(oc_uuid_t *subject, oc_sec_ace_t *ace,
                          oc_obt_device_status_cb_t cb, void *data);
 void oc_obt_free_ace(oc_sec_ace_t *ace);
 
+void oc_obt_remove_credential_by_credid(size_t device, long credid);
+
 #ifdef __cplusplus
 }
 #endif
diff --git a/security/oc_obt.c b/security/oc_obt.c
index b8e744a..7b2adfd 100644
--- a/security/oc_obt.c
+++ b/security/oc_obt.c
@@ -1395,4 +1395,15 @@ oc_obt_shutdown(void)
     device = (oc_device_t *)oc_list_pop(oc_devices);
   }
 }
+
+void
+oc_obt_remove_credential_by_credid(size_t device, long credid)
+{
+  oc_sec_cred_t *cred = oc_sec_get_cred_by_credid(credid, device);
+  if (cred)
+  {
+    oc_sec_remove_cred(cred, device);
+  }
+}
+
 #endif /* OC_SECURITY */
-- 
2.16.1.windows.1

